Updating to Swift 3

tmspzz opened the pull request:
This pull request makes Tyro compatible with Swift 3

Edit: I tried to get travis working but I don't know enough to fix it :(

tmspzz committed:
Updating to Swift 3

tmspzz committed:
Updating Travis

CodaFi commented on .travis.yml:
```diff
@@ -1,5 +1,5 @@
 language: objective-c
-osx_image: xcode7.3
+osx_image: xcode8.0
```
This is not enough to get it building.  The old scripts directory needs to go and this entire file needs to be replaced by a [more modern script](https://github.com/typelift/Concurrent/blob/master/.travis.yml).

CodaFi commented on Carthage/Checkouts/SwiftCheck:
```diff
@@ -1 +1 @@
-Subproject commit f2141461f015315cc583222f0bb30ae07da8aad8
+Subproject commit 983b5954f221d36efe3469e8a5155d953034994a
```
Please don't manually update the submodules.  Edit the cartfile and run `carthage update --use-submodules` then commit the result.

CodaFi commented on Carthage/Checkouts/Swiftz:
```diff
@@ -1 +1 @@
-Subproject commit 78665a0921857ce6b1408bd91a5069931cab0191
+Subproject commit afe8639115abd249c85a477b9dea1f0f8c299815
```
Ditto

CodaFi commented on Tyro/Error.swift:
```diff
@@ -8,26 +8,26 @@
     case Array([JSONError])           // array of errors
     case TypeMismatch(String, String) // expected / actual
-    case Error(ErrorType, String)     // error / message
+    case Error(Error, String)     // error / message
```
Canonical case names for Swift 3 structures begin with a lowercase letter.  That means `Error` should not be a case name.

tmspzz replied:
You mean lowercase enum cases?

I saw that in Swiftx (the swift 3 version) Either still has uppercase members for left and right.  I though since these really wanted to indicate types somehow (Array, Object, String ...) the exception would be acceptable. Looked a little odd when I tried. Will do anyways.

CodaFi commented on Tyro/NSDataExt.swift:
```diff
@@ -9,9 +9,9 @@
 import Foundation
 import Swiftz
 
-extension NSData {
```
Please remove this file and inline the definition of this extension.

CodaFi commented on TyroTests/JSONValueSpec.swift:
```diff
@@ -13,61 +13,67 @@ import SwiftCheck
-				return Gen.pure(JSONValue.Null)
-			}
-		}
+        let t: Gen<UInt> = UInt.arbitrary
```
The indentation of this entire file has changed.

CodaFi reviewed (changes requested):
Generally, the codes style in this framework needs to be fixed.  We always leave space around colons that indicate types and spaces after but not before colons that indicate parameters.

CodaFi commented:
Thank you for working on this.  Still, I can't help but think this entire framework can be [replaced by a tweet](https://gist.github.com/griotspeak/dc41f9c2a84d50592690d01c0043aa1e).

tmspzz committed:
Specifying swift version for Tyro-iOS

tmspzz committed:
Updating dependencies

Commit:
Use correct version of SwiftCheck

Commit:
Specify swift version 3.0

Commit:
Changed travis CI image

Commit:
Update quickly readme

Commit:
Remove notifications when travis-ci runs

Commit:
Automate release of Cocoapods spec on tag

https://fuller.li/posts/automated-cocoapods-releases-with-ci/

Commit:
Remove warning

Commit:
Update Podspec

Note will failed until Swiftz publish its spec. See https://github.com/typelift/Swiftz/issues/327

Commit:
Disable cocoapods for now
